pkg/mmap/mmap_unix.go

```diff
@@ -10,15 +10,17 @@ package mmap
 import (
 	"os"
 	"syscall"
+
+	errors2 "github.com/influxdata/influxdb/pkg/errors"
 )
 
 // Map memory-maps a file.
-func Map(path string, sz int64) ([]byte, error) {
+func Map(path string, sz int64) (data []byte, err error) {
 	f, err := os.Open(path)
 	if err != nil {
 		return nil, err
 	}
-	defer f.Close()
+	defer errors2.Capture(&err, f.Close)()
```

Comment:
oops - I didn't catch this the first time, but `err` is actually in the `:=` on line 18. This still works (since `:=` here doesn't shadow err due to a go quirk) but is more confusing than it should be.